### PR TITLE
[ACCEL]Borovenskiy/bug/accel 553/hide entry when there are no managers in the cohort

### PR DESCRIPTION
**Pull request by NikolayBorovenskiy**
**Description:** PR fixes several bugs at the same time. A
ll bugs are very similar to each other and it was convenient to make such a union.
Many bugs fixed in the feature of adding cohort leaders when creating and editing a cohort.

**Youtrack:** https://youtrack.raccoongang.com/issue/ACCEL-555
https://youtrack.raccoongang.com/issue/ACCEL-553
https://youtrack.raccoongang.com/issue/ACCEL-556

**Reviewers:**
- @idegtiarov 
- @Golub-Sergey 

**Post merge:**
- [+] Delete working branch (if not needed anymore)

**Author concerns:** PR contains custom migration. It is necessary and safe. Locally I checked and everything works reliably, but still, we need to be careful.

### CODE REVIEW

File: lms/djangoapps/instructor/views/api.py [hunk 1]
```diff
@@ -3391,13 +3414,15 @@ def update_cohort_assignment(request, course_id):
         cohort_for_change = [cohort_id]
 
     if is_assignment:
-        cohort_to_incert = [CohortAssigment(cohort_id=cohort, user=user) for cohort in cohort_for_change]
-        CohortAssigment.objects.bulk_create(cohort_to_incert)
-    else:
-        CohortAssigment.objects.filter(cohort__id__in=cohort_for_change, user=user).delete()
+        created = False
+        for cohort in cohort_for_change:
+            if not cohort:
```
Comment (idegtiarov) [inline lms/djangoapps/instructor/views/api.py]: Let's avoid it by not adding None objects into `cohort_for_change` list.

Comment (NikolayBorovenskiy) [reply lms/djangoapps/instructor/views/api.py]: Add a fix. Thanks to that you mention it to me.

**Comment by NikolayBorovenskiy** [comment]
@idegtiarov if you do not mind let's merge the PR because I need the code form it for the task https://youtrack.raccoongang.com/issue/ACCEL-554

**Comment by idegtiarov** [comment]
@NikolayBorovenskiy Please resolve conflicts so we will be able to merge PR

**Pull request merged**